src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ManagedIndexCoordinator.kt

```diff
@@ -256,57 +252,35 @@ class ManagedIndexCoordinator(
 
     @OpenForTesting
     suspend fun sweepClusterChangedEvent(event: ClusterChangedEvent) {
-        val indicesDeletedRequests = event.indicesDeleted()
-                    .filter { event.previousState().metadata().index(it)?.getPolicyID() != null }
-                    .map { deleteManagedIndexRequest(it.uuid) }
-
-        /*
-        * Update existing indices that have added or removed policy_ids
-        * There doesn't seem to be a fast way of finding indices that meet the above conditions without
-        * iterating over every index in cluster state and comparing current policy_id with previous policy_id
-        * If this turns out to be a performance bottle neck we can remove this and enforce
-        * addPolicy/removePolicy API usage for existing indices and let the background sweep pick up
-        * any changes from users that ignore and use the ES settings API
-        * */
-        var hasCreateRequests = false
-        val updateManagedIndicesRequests = mutableListOf<DocWriteRequest<*>>()
-        val indicesWithPolicyID = mutableListOf<String>()
-        val indicesToRemoveManagedIndexMetaDataFrom = mutableListOf<Index>()
-        event.state().metadata().indices().forEach {
-            val previousIndexMetaData = event.previousState().metadata().index(it.value.index)
-            val policyID = it.value.getPolicyID()
-            val request: DocWriteRequest<*>? = when {
-                it.value.shouldCreateManagedIndexConfig(previousIndexMetaData) && policyID != null -> {
-                    indicesWithPolicyID.add(it.value.index.name)
-                    hasCreateRequests = true
-                    managedIndexConfigIndexRequest(it.value.index.name, it.value.indexUUID, policyID, jobInterval)
-                }
-                it.value.shouldDeleteManagedIndexConfig(previousIndexMetaData) ->
-                    deleteManagedIndexRequest(it.value.indexUUID)
-                else -> null
-            }
-
-            if (request != null) updateManagedIndicesRequests.add(request)
-
-            if (it.value.shouldDeleteManagedIndexMetaData()) indicesToRemoveManagedIndexMetaDataFrom.add(it.value.index)
+        // deal with indices delete event
+        var removeManagedIndexReq = emptyList<DocWriteRequest<*>>()
+        var indicesToClean = emptyList<Index>()
+        if (event.indicesDeleted().isNotEmpty()) {
+            val deletedIndices = event.indicesDeleted().map { Index(it.name, it.uuid) }
```

Comment:
Why do we need this if event.indicesDeleted is already a List<Index>?

Reply:
List\<Index\>*

Reply:
lol, I don't know. change to just use event.indicesDeleted